src/auth.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Returns the Basic auth string based on the `username` and `password` parts of the given URL.
+ * If the URL does not contain neither username nor password, returns `null`.
+ * @param url URL object to process
+ * @returns `Basic BASE64` string
+ */
+export function getBasic(url: URL): string | null {
```

Comment:
Could we please call this something more descriptive? `buildBasicAuthHeader` for instance?